File: CS/App_Code/Params.cs
using System;
using System.Collections.Generic;
using System.Collections.Specialized;
using System.Linq;
using System.Web;
using System.Web.SessionState;

public static class CallbackParams {
    public static NameValueCollection Get(HttpRequest request) {
        String __CALLBACKPARAM = Convert.ToString(request.Form["__CALLBACKPARAM"]);

        if (__CALLBACKPARAM != null && __CALLBACKPARAM.IndexOf("http://tempuri.org/?") > 0) {

            if (__CALLBACKPARAM.EndsWith(";")) {
                __CALLBACKPARAM = __CALLBACKPARAM.Remove(__CALLBACKPARAM.Length - 1);
            }

            __CALLBACKPARAM = __CALLBACKPARAM.Substring(
                            __CALLBACKPARAM.IndexOf("http://tempuri.org/?"));

            NameValueCollection parameters =
                        System.Web.HttpUtility.ParseQueryString(new Uri(__CALLBACKPARAM).Query);

            return parameters;

        }

        return null;
    }
}

public static class SqlParams {
    public static NameValueCollection Get(NameValueCollection parameters, HttpSessionState session) {

        if (parameters != null) {             

            String sort = parameters["Sort"];
            if (String.IsNullOrWhiteSpace(sort)) {
                sort = Convert.ToString(session["Sort"]);
            }

            if (String.Equals(sort, "title", StringComparison.InvariantCultureIgnoreCase)) {
                sort = "Title";
            } 
            else {
                sort = "Chapter";
            }

            parameters["Sort"] = sort;

            String mode = parameters["Mode"];
            if (String.IsNullOrWhiteSpace(mode)) {
                mode = Convert.ToString(session["Mode"]);
            }

            if (String.Equals(mode, "desc", StringComparison.InvariantCultureIgnoreCase)) {
                mode = "DESC";
            } else {
                mode = "ASC";
            }

            parameters["Mode"] = mode;

        }

        return parameters;
    }
}

File: VB/App_Code/Params.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Collections.Specialized
Imports System.Linq
Imports System.Web
Imports System.Web.SessionState

Public NotInheritable Class CallbackParams
	Private Sub New()
	End Sub
	Public Shared Function [Get](ByVal request As HttpRequest) As NameValueCollection
		Dim __CALLBACKPARAM As String = Convert.ToString(request.Form("__CALLBACKPARAM"))

		If __CALLBACKPARAM IsNot Nothing AndAlso __CALLBACKPARAM.IndexOf("http://tempuri.org/?") > 0 Then

			If __CALLBACKPARAM.EndsWith(";") Then
				__CALLBACKPARAM = __CALLBACKPARAM.Remove(__CALLBACKPARAM.Length - 1)
			End If

			__CALLBACKPARAM = __CALLBACKPARAM.Substring(__CALLBACKPARAM.IndexOf("http://tempuri.org/?"))

			Dim parameters As NameValueCollection = System.Web.HttpUtility.ParseQueryString(New Uri(__CALLBACKPARAM).Query)

			Return parameters

		End If

		Return Nothing
	End Function
End Class

Public NotInheritable Class SqlParams
	Private Sub New()
	End Sub
	Public Shared Function [Get](ByVal parameters As NameValueCollection, ByVal session As HttpSessionState) As NameValueCollection

		If parameters IsNot Nothing Then

			Dim sort As String = parameters("Sort")
			If String.IsNullOrWhiteSpace(sort) Then
				sort = Convert.ToString(session("Sort"))
			End If

			If String.Equals(sort, "title", StringComparison.InvariantCultureIgnoreCase) Then
				sort = "Title"
			Else
				sort = "Chapter"
			End If

			parameters("Sort") = sort

			Dim mode As String = parameters("Mode")
			If String.IsNullOrWhiteSpace(mode) Then
				mode = Convert.ToString(session("Mode"))
			End If

			If String.Equals(mode, "desc", StringComparison.InvariantCultureIgnoreCase) Then
				mode = "DESC"
			Else
				mode = "ASC"
			End If

			parameters("Mode") = mode

		End If

		Return parameters
	End Function
End Class